solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/SolrFeature.java

```diff
@@ -238,13 +238,12 @@ public DocIdSetIterator getDocIdSetIteratorFromQueries(List<Query> queries,
 
     public class SolrFeatureScorer extends FeatureScorer {
       final private Scorer solrScorer;
-      final private DocIdSetIterator itr;
 
       public SolrFeatureScorer(FeatureWeight weight, Scorer solrScorer,
           DocIdSetIterator filterIterator) {
-        super(weight);
+        super(weight, filterIterator);
         this.solrScorer = solrScorer;
-        itr = new SolrFeatureScorerIterator(filterIterator,
+        this.itr =  new SolrFeatureScorerIterator(filterIterator,
```

Comment:
The itr that needs to be used for super is this one, not the one passed in the constructor.